src/Agent.Listener/Configuration/ConfigurationProvider.cs

```diff
@@ -47,43 +51,57 @@ public override void Initialize(IHostContext hostContext)
             _agentServer = HostContext.GetService<IAgentServer>();
         }
 
-        public void UpdateAgentSetting(AgentSettings settings)
+        public void ReadSettings(AgentSettings settings)
         {
-            // No implementation required
         }
 
         public string GetServerUrl(CommandSettings command)
         {
             return command.GetUrl();
         }
 
-        public async Task<int> GetPoolId(CommandSettings command)
+        public async Task<AgentSettings> GetPoolId(CommandSettings command)
         {
             int poolId = 0;
             string poolName;
 
             poolName = command.GetPool();
             poolId = await GetPoolIdAsync(poolName);
             Trace.Info($"PoolId for agent pool '{poolName}' is '{poolId}'.");
- 
-            return poolId;
+
+            return new AgentSettings() { PoolId = poolId};
         }
 
         public string GetFailedToFindPoolErrorString() => StringUtil.Loc("FailedToFindPool");
 
-        public Task<TaskAgent> UpdateAgentAsync(int poolId, TaskAgent agent, CommandSettings command)
+        public string GetAgentWithSameNameAlreadyExistErrorString(AgentSettings agentSettings)
```

Comment:
just let this method throw the exception, seems weird to get back an error string. :)

public void ThrowTaskAgentExistException()
{

}